ecl/hqlcpp/hqlttcpp.cpp

```diff
@@ -6031,6 +6047,33 @@ IHqlExpression * WorkflowTransformer::extractWorkflow(IHqlExpression * untransfo
         if ((info.persistOp != no_persist) && expr->isAction())
             setValue.setown(transformSequentialEtc(setValue));
 
+         if(info.persistOp == no_critical)
```

Comment:
Some of the code within this if isn't needed.  - e.g., all of checkArgs.  I suspect criticalWfid and others.